agent: Fix endpoint restore with unmounted BPF filesystem

tgraf opened the pull request:
The function Daemon.restoreOldEndpoints() had errored out when the endpoint BPF
map could not be opened. This would prevent restoring of endpoints when the BPF
filesystem was not mounted. The error does not matter if we can't open the map
as the map will be initialized from scratch and no cleaning up of old
potentially stale entries is necessary.

Fixes: 4ecf111c35f ("agent: Fix temporary corruption of BPF endpoint map on restart")

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6624)
<!-- Reviewable:end -->

raybejjani commented on daemon/state.go:
```diff
@@ -144,21 +144,25 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
-		delete(existingEndpoints, ep.IPv4.String())
-		delete(existingEndpoints, ep.IPv6.String())
+		if existingEndpoints != nil {
+			delete(existingEndpoints, ep.IPv4.String())
```
This should be safe on a nil map (I just tested https://play.golang.org/p/rr9va0dF2kp), but perhaps you wanted to make it clear that it did nothing when nil?

tgraf replied:
Yeah, it's more of a documentation purpose to make it clear that existingEndpoints is expected to be nil in some cases.

raybejjani commented on daemon/state.go:
```diff
@@ -144,21 +144,25 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
-			} else {
-				log.Debugf("Removed outdated endpoint %d from endpoint map", info.LxcID)
+	if existingEndpoints != nil {
+		for hostIP, info := range existingEndpoints {
```
This loop should no-op when the map is nil, but perhaps you wanted to make it clear that it did nothing when nil?

raybejjani reviewed (approved):
I'm probably not the best person to review this.

tgraf committed:
agent: Fix endpoint restore with unmounted BPF filesystem

The function Daemon.restoreOldEndpoints() had errored out when the endpoint BPF
map could not be opened. This would prevent restoring of endpoints when the BPF
filesystem was not mounted. The error does not matter if we can't open the map
as the map will be initialized from scratch and no cleaning up of old
potentially stale entries is necessary.

Fixes: 4ecf111c35f ("agent: Fix temporary corruption of BPF endpoint map on restart")

Signed-off-by: Thomas Graf <thomas@cilium.io>
